zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbRoleState.java

```diff
@@ -68,6 +69,20 @@ public void createRole(final RoleRecord roleRecord) {
     roleByNameColumnFamily.insert(roleName, fkRoleKey);
   }
 
+  @Override
+  public void addEntity(final RoleRecord roleRecord) {
+    roleKey.wrapLong(roleRecord.getRoleKey());
+    entityKey.wrapLong(roleRecord.getEntityKey());
+    entityTypeValue.setEntityType(roleRecord.getEntityType());
+    entityTypeByRoleColumnFamily.insert(fkRoleKeyAndEntityKey, entityTypeValue);
+
+    // update role record
+    final var persistedRole = roleColumnFamily.get(roleKey);
+    persistedRole.getRole().setEntityKey(roleRecord.getEntityKey());
+    persistedRole.getRole().setEntityType(roleRecord.getEntityType());
+    roleColumnFamily.update(roleKey, persistedRole);
```

Comment:
💭 I'm not sure if this update is needed, it might be useful

Reply:
❌ We don't lets remove it. I already mentioned this in the other PR you created but now I'm certain we should change it. Let's define the `roleKey` and the `roleName` in the `PersistedRole` with properties, instead of one property with the `RoleRecord`. That way we don't store the `entityKey` and the `entityType` in there.